### ISSUE TITLE
Racy Spec: ActorSystemSpec.Reliable_deny_creation_of_actors_while_shutting_down

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#1762]
Starting failing consistently now - this is a recent failure.

```
Assert.Empty() Failure
   at Akka.Tests.Actor.ActorSystemSpec.Reliable_deny_creation_of_actors_while_shutting_down() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka.Tests\Actor\ActorSystemSpec.cs:line 195
« Hide stacktrace
```

Appears to have started after the Akka.Persistence 2.4 changes were merged in CC @cconstantin

**Comment by Aaronontheweb** [comment]
Full trace: http://petabridge-ci.cloudapp.net/viewLog.html?buildId=6993&buildTypeId=AkkaNet_AkkaNetWindowsUnitTests&guest=1

**Comment by cconstantin** [comment]
Hmm, not sure why changes in persistence would affect this.

**Comment by Aaronontheweb** [comment]
@cconstantin yeah me neither, but that's just when this issue started showing up. I'm looking into it now and it looks like there's some technical debt around how we do shutdowns that needs to be paid off.

**Comment by Aaronontheweb** [comment]
#1766 explains more of the story behind this issue

**Comment by JeffCyr** [comment]
The test fails because there is a race condition between ActorSystem.ActorOf and the handling of the Terminate message in the UserGuardian.

There is a difference in the way actors are created under the UserGuardian in Akka.net and Akka jvm. The jvm implementation uses AttachChild to create a RepointableActorRef.
https://github.com/akka/akka/blob/0d8dd3c7114ed5258208aaed3575460a056876bb/akka-actor/src/main/scala/akka/actor/ActorSystem.scala#L588

I changed the ActorSystem to behave like the JVM, the race condition is now less common, but I am still able to reproduce sometimes (without debugger attached only)...

I'm still trying to find the remaining race condition, I'll update this issue when I have progressed.

**Comment by JeffCyr** [comment]
FYI: Using AttachChild fixed the issue, I was still reproducing the failure because I didn't have the fix from PR #1978.

The fix was put on hold for now until we measure the impact of the change. It would be a subtle breaking change to return RepointableActorRef in Actorsystem.ActorOf.

**Comment by Aaronontheweb** [comment]
Related to #2024

**Comment by Aaronontheweb** [comment]
Resolved by #2086
